### ISSUE TITLE
babel-parser: Introduce error code

### ISSUE THREAD

**Issue by sosukesuzuki** [babel/babel#13007]
## Feature Request

- [x] I would like to work on this feature!

**Is your feature request related to a problem?**

Yes, Prettier needs to handle the errors provided by babel-parser in recovery mode, which can be ignored.

The current  Prettier hardcodes error messages for this purpose. See https://github.com/prettier/prettier/blob/3995c2a1a1839753e08f0576c4161876bc5ad62e/src/language-js/parser-babel.js#L147-L204. This is not a good. Fixing Babel's error messages may break Prettier.

Context: https://github.com/prettier/prettier/pull/10495#issuecomment-796499107

**Describe the solution you'd like**

Introduce error code.

**Describe alternatives you've considered**.

Continue hardcoding...

**Documentation, Adoption, Migration Strategy**

I think the following interface looks good.

```js
const { parse } = require("@babel/parser");

const ast = parse("const foo = 1; const foo = 2;", { errorRecovery: true }));

console.log(ast.errors[0].message); //  Identifier 'foo' has already been declared
console.log(ast.errors[0].code); // BABEL_PARSER_SYNTAX_ERROR_001

```

**Comment by nicolo-ribaudo** [comment]
I would prefer something like `ast.errors[0].code === "VarRedeclaration"`, reusing the error codes we already use internally to raise errors (defined at https://github.com/babel/babel/blob/main/packages/babel-parser/src/parser/error-message.js).

**Comment by nicolo-ribaudo** [comment]
Implemented in 7.14.0
